src/Math-Statistics/Collection.extension.st

```diff
@@ -0,0 +1,37 @@
+Extension { #name : #Collection }
+
+{ #category : #'*Math-Statistics' }
+Collection >> geometricMean [
+	"Answer with the geometric mean of the collection"
+	"#(1 1 2 3 4 5 5 6 6 7 8 9) geometricMean  >>> 3.8583980015011208"
+	"{ 4. 1.  1 / 32} geometricMean >>> (1/2)"
+	"#(3.14 1 4.56 0.333) geometricMean >>> 1.4776945822943937"
+
+	^(self reduce: [ :a :b | a * b ] ) raisedToFraction:  1 / self size.
```

Comment:
Note that this may result in Float overflow on large collections.
An alternative is to sum the logarithms.   `(a*b*c)^(1/3)=exp(ln(a*b*c))^(1/3)=exp((ln(a)+ln(b)+ln(c))/3)`
That is something like `^((self collect: #ln) sum / self size) exp`.
The price is that we may answer an inexact Float instead of an exact Integer in some edge cases.
For example`{1/3. 2/3. 8/3. 16/3} geometricMean` would answer 1.3333333333333333 instead of (4/3)...
Also note that the geometricMean may not be well defined for negative numbers, likewise the harmonicMean...

Reply:
Thanks for pointing this out. I have updated the implementation with your suggestion and added a guard clause similar to harmonicMean.